app/code/Magento/QuoteGraphQl/Model/Cart/AddGroupedProductToCart.php

```diff
@@ -0,0 +1,156 @@
+     * @param ProductRepositoryInterface $productRepository
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    public function __construct(
+        ArrayManager $arrayManager,
+        DataObjectFactory $dataObjectFactory,
+        ProductRepositoryInterface $productRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder
+    ) {
+        $this->arrayManager = $arrayManager;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->productRepository = $productRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    }
+
+    /**
+     * Add simple product to cart
+     *
+     * @param Quote $cart
+     * @param array $cartItemData
+     * @return void
+     * @throws GraphQlNoSuchEntityException
+     * @throws GraphQlInputException
+     */
+    public function execute(Quote $cart, array $cartItemData): void
+    {
+        $sku = $this->extractSku($cartItemData);
+        $subProducts = $this->extractSubProducts($cartItemData);
+
+        try {
+            $product = $this->productRepository->get($sku);
+        } catch (NoSuchEntityException $e) {
+            throw new GraphQlNoSuchEntityException(__('Could not find a product with SKU "%sku"', ['sku' => $sku]));
+        }
+
+        $result = $cart->addProduct($product, $this->createBuyRequest($product, $subProducts));
+
+        if (is_string($result)) {
+            throw new GraphQlInputException(__($result));
+        }
+    }
+
+    /**
+     * Extract SKU from cart item data
+     *
+     * @param array $cartItemData
+     * @return string
+     * @throws GraphQlInputException
+     */
+    private function extractSku(array $cartItemData): string
+    {
+        $sku = $this->arrayManager->get('data/sku', $cartItemData);
+        if (!isset($sku)) {
+            throw new GraphQlInputException(__('Missing key "sku" in cart item data'));
+        }
+        return (string)$sku;
+    }
+
+    /**
+     * Extract Customizable Options from cart item data
+     *
+     * @param array $cartItemData
+     * @return array
+     */
+    private function extractSubProducts(array $cartItemData): array
+    {
+        $subProducts = $cartItemData['data']['grouped_products'];
+
+        $subProductsData = [];
+        foreach ($subProducts as $subProduct) {
+            $subProductsData[$subProduct['sku']] = $subProduct['qty'];
+        }
+
+        return $subProductsData;
+    }
+
+    /**
+     * Format GraphQl input data to a shape that buy request has
+     *
+     * @param ProductInterface $product
+     * @param array $subProducts
+     * @return DataObject
+     */
+    private function createBuyRequest(ProductInterface $product, array $subProducts): DataObject
+    {
+        $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter('sku', array_keys($subProducts) , 'in')
+            ->create();
+
+        $products = $this->productRepository->getList($searchCriteria)->getItems();
```

Comment:
We can use `$procudctCollection->getData()` for performance improvement
We don't need `whole` product object, just SKUs